app/actors/JvmMetricsCollector.java

```diff
@@ -15,42 +15,69 @@
  */
 package actors;
 
+import akka.actor.ActorSystem;
 import akka.actor.Cancellable;
 import akka.actor.UntypedActor;
+import akka.dispatch.Dispatcher;
 import com.arpnetworking.logback.annotations.LogValue;
 import com.arpnetworking.metrics.MetricsFactory;
+import com.arpnetworking.metrics.jvm.ExecutorServiceMetricsRunnable;
 import com.arpnetworking.metrics.jvm.JvmMetricsRunnable;
 import com.arpnetworking.play.configuration.ConfigurationHelper;
 import com.arpnetworking.steno.LogValueMapFactory;
 import com.arpnetworking.steno.Logger;
 import com.arpnetworking.steno.LoggerFactory;
 import com.google.common.base.MoreObjects;
+import com.google.common.collect.Maps;
 import com.google.inject.Inject;
 import play.Configuration;
+import scala.concurrent.ExecutionContextExecutor;
 import scala.concurrent.duration.FiniteDuration;
+import scala.concurrent.forkjoin.ForkJoinPool;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ForkJoinTask;
+import java.util.concurrent.Future;
+import java.util.concurrent.RunnableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
 
 /**
  * Actor responsible for collecting JVM metrics on a periodic basis.
  *
  * @author Deepika Misra (deepika at groupon dot com)
+ * @author Ville Koskela (ville dot koskela at inscopemetrics dot com)
  */
 public final class JvmMetricsCollector extends UntypedActor {
 
     /**
      * Public constructor.
      *
      * @param configuration Play app configuration.
+     * @param actorSystem Akka <code>ActorySystem</code>.
      * @param metricsFactory An instance of <code>MetricsFactory</code>.
      */
     @Inject
     public JvmMetricsCollector(
             final Configuration configuration,
+            final ActorSystem actorSystem,
```

Comment:
We can get the Actor's ActorSystem from context().system().  Is there a special reason you're injecting it?